### PR TITLE
Remove the use of euclid types from the software renderer public API

### LINKED ISSUE #1538: Software Renderer API cleanups
In the API review we decided on a few changes to the software renderer types:

- Remove `PhysicalSize`/`PhysicalLength`/`PhysicalPx`, always use usize
- Change `new`:
  * `fn new(window_adapter: &Weak<WindowAdapter>, dirty_tracking_policy: DirtyTracking) -> SoftwareRenderer`
  * Remove `window` paramter from `render_XXX` functions.

In the LineBufferProvider:

Assumption: The LineBufferProvider operates on a data structure that can be represented as a slice of the associated target pixel type.

- Change `process_line`'s `range` to be a `pixel_range: Range<usize>`. 
- Change `line` parameter to `line: usize`.

### PR DESCRIPTION
**Pull request by tronical**
Closes #1538

### CODE REVIEW

File: examples/mcu-board-support/stm32h735g.rs [hunk 1]
```diff
@@ -324,7 +323,7 @@ impl slint::platform::Platform for StmBackend {
             if let Some(window) = self.window.borrow().clone() {
                 window.draw_if_needed(|renderer| {
                     while layer.is_swap_pending() {}
-                    renderer.render(work_fb, PhysicalLength::new(DISPLAY_WIDTH as i16));
+                    renderer.render(work_fb, DISPLAY_WIDTH as usize);
```
Comment (ogoffart) [inline examples/mcu-board-support/stm32h735g.rs:326]:
```suggestion
                    renderer.render(work_fb, DISPLAY_WIDTH);
```

File: internal/backends/winit/renderer/sw.rs [hunk 2]
```diff
@@ -51,16 +50,15 @@ impl<const BUFFER_COUNT: usize> super::WinitCompatibleRenderer for SoftwareRende
...
                 ) {
-                    let len = (range.end.get() - range.start.get()) as usize;
-                    let line_begin = line.get() as usize * self.line.len();
+                    let len = range.end - range.start;
+                    let line_begin = line * self.line.len();
                     let sub = &mut self.line[..len];
                     render_fn(sub);
-                    for (dst, src) in self.buffer[line_begin + (range.start.get() as usize)
-                        ..line_begin + (range.end.get() as usize)]
+                    for (dst, src) in self.buffer[line_begin + range.start..line_begin + range.end]
```
Comment (ogoffart) [inline internal/backends/winit/renderer/sw.rs:61]: Not sure about this one:

```suggestion
                    for (dst, src) in self.buffer[line_begin..][range]
```

File: internal/backends/winit/renderer/sw.rs [hunk 3]
```diff
@@ -51,16 +50,15 @@ impl<const BUFFER_COUNT: usize> super::WinitCompatibleRenderer for SoftwareRende
...
                     &mut self,
-                    line: PhysicalLength,
-                    range: core::ops::Range<PhysicalLength>,
+                    line: usize,
+                    range: core::ops::Range<usize>,
                     render_fn: impl FnOnce(&mut [Self::TargetPixel]),
                 ) {
-                    let len = (range.end.get() - range.start.get()) as usize;
-                    let line_begin = line.get() as usize * self.line.len();
+                    let len = range.end - range.start;
```
Comment (ogoffart) [inline internal/backends/winit/renderer/sw.rs:57]:
```suggestion
                    let len = range.len();
```

And maybe even no need to have a variable for that then.

**Review by ogoffart — APPROVED** [review]
Yes, this is a clear imrovement.

I think you'll also have to fix some doctests

**Comment by ogoffart** [comment]
I wonder why the doctest for `render_by_line`  passed the CI.  
could it be because the swrenderer feature is not enabled?  
We pass `--all-features`  to the cargo build, but not to the cargo test, maybe that's why

File: internal/backends/winit/renderer/sw.rs (hunk 3 above)
Comment (tronical) [reply internal/backends/winit/renderer/sw.rs:57]: Good idea

**Pull request merged**